src/cephadm/cephadmlib/daemons/smb.py

```diff
@@ -201,6 +267,77 @@ def args(self) -> List[str]:
         return super().args() + ['update-config', '--watch']
 
 
+class CTDBMigrateInitContainer(SambaContainerCommon):
+    def name(self) -> str:
+        return 'ctdbMigrate'
+
+    def args(self) -> List[str]:
+        # TODO: not only do we want to only do this on node 0, we only
+        # want to do it exactly ONCE per cluster even on pnn 0. This needs
+        # additional work to get that right.
+        return super().args() + [
+            '--skip-if=env:NODE_NUMBER!=0',
+            'ctdb-migrate',
+            '--dest-dir=/var/lib/ctdb/persistent',
+            '--archive=/var/lib/samba/.migrated',
+        ]
+
+
+class CTDBMustHaveNodeInitContainer(SambaContainerCommon):
+    def name(self) -> str:
+        return 'ctdbMustHaveNode'
+
+    def args(self) -> List[str]:
+        args = super().args()
+        unique_name = self.cfg.identity.daemon_name
+        args += [
+            'ctdb-must-have-node',
+            # hostname is a misnomer (todo: fix in sambacc)
+            f'--hostname={unique_name}',
+            '--take-node-number-from-env',
+        ]
+        return args
+
+
+class CTDBDaemonContainer(SambaContainerCommon):
+    def name(self) -> str:
+        return 'ctdbd'
+
+    def args(self) -> List[str]:
+        return super().args() + [
+            'run',
+            'ctdbd',
+            '--setup=smb_ctdb',
+            '--setup=ctdb_config',
+            '--setup=ctdb_etc',
+        ]
+
+    def container_args(self) -> List[str]:
+        cargs = super().container_args()
+        # make conditional?
+        # CAP_NET_ADMIN is needed for event script to add public ips to iface
```

Comment:
With the current set of changes do we support assigning public ip addresses? Are we adding `CAP_NET_ADMIN` for future use?

In the absence of an empty public addresses list do we see any "not a public address" log entries when clients connections are made against node IPs?

Reply:
Future use case. Hopefully my next PR, but I was experimeting with it manually using this PR and I didn't see the point to remove that flag just to add it back in the next pr.